shared/middleware/helpers/bucket_helper.rb

```diff
@@ -87,6 +123,21 @@ def create_or_replace(encrypted_channel_id, filename, body, version = nil, abuse
     response
   end
 
+  #
+  # Copy an object within a channel, creating a new object in the channel.
+  #
+  # @param [String] encrypted_channel_id - App-identifying token
+  # @param [String] filename - Destination name for new object
+  # @param [String] source_filename - Name of object to be copied
+  # @param [Hash] S3 response from copy operation
+  def copy(encrypted_channel_id, filename, source_filename)
+    owner_id, channel_id = storage_decrypt_channel_id(encrypted_channel_id)
+
+    key = s3_path owner_id, channel_id, filename
+    copy_source = @bucket + '/' + s3_path(owner_id, channel_id, source_filename)
+    @s3.copy_object(bucket: @bucket, key: key, copy_source: copy_source)
```

Comment:
Oh, interesting.  I'm actually doing the check for whether the source object exists up a layer, in the `copy_file` method of files_api.rb when I check the app quota limits, but it's a good idea to catch that case at this layer too.  I wonder if we need to add some tests directly around `BucketHelper` to catch this sort of thing (I have an animations API-level test for a copy operation with a missing source).

I'll add a TODO for the abuse metadata and a Pivotal item.  I imagine we'll end up following the pattern we have for assets.